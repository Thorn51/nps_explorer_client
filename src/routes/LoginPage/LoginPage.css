.alert {
  color: red;
  font-weight: bolder;
  text-align: center;
  padding: 15px;
}

.demo_user {
  padding: 15px;
}

.credentials {
  padding-top: 10px;
}
