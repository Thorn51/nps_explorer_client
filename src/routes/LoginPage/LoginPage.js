import React from "react";
import Navigation from "../../components/Navigation/Navigation";
import Hero from "../../components/Hero/Hero";
import "./LoginPage.css";

export default function LoginPage() {
  return (
    <>
      <Navigation />
      <Hero
        title="National Parks Explorer"
        tag="Login"
        background="https://www.nps.gov/npgallery/GetAsset/34F574D2-1DD8-B71B-0BDE3EA69DD198E8/proxy/hires"
      />
      <main>
        <form class="login_form">
          <label for="login_email">Email:</label>
          <input
            type="text"
            placeholder="Enter Email"
            name="email"
            id="login_email"
            required
          />
          <label for="login_password">Password:</label>
          <input
            type="password"
            placeholder="Enter Password"
            name="password"
            id="login_password"
            required
          />
          <button type="submit" class="form_button">
            Login
          </button>
        </form>
        <div class="form_redirect">
          <hr />
          <p>
            Do you need to create an account? Visit the
            <a href="#registration">registration page</a>.
          </p>
        </div>
      </main>
    </>
  );
}
