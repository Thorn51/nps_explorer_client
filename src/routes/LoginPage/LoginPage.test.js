import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import LoginPage from "./LoginPage";

describe("<LoginPage />", () => {
  it("Renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<LoginPage />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("Renders the UI as expected", () => {
    const tree = renderer.create(<LoginPage />).toJSON();
    expect("tree").toMatchSnapshot();
  });
});
