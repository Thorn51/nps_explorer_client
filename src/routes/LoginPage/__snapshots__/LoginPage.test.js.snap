// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginPage /> Renders the UI as expected 1`] = `"tree"`;
